### ISSUE TITLE
Misleading step method in lr_scheduler.ReduceLROnPlateau

### ISSUE THREAD

**Issue by LeviViana** [pytorch/pytorch#12146]
In short, since ReduceLROnPlateau takes an optimizer as argument, one can simply assume that it wraps the optimizer and the step method from the ReduceLROnPlateau object will actually call the step method of the optimizer. 

This kind of behaviour can be expected from a pytorch user point of view, "classes wrapping classes" just happens all the time.

There isn't even any warning in the documentation telling the user that when you call the step method no backprop is going to happen although you passed an optimizer as argument and you are explicitly calling a "step method".

Alternatives:
<li>simply put a WARNING in the documentation
<li>implement a <i>optimizer_step</i> method to explicitly allow the user to call the optimizer step method from the ReduceLROnPlateau object

I actually didn't carefully read the documentation the first time I used ReduceLROnPlateau. So I assumed that I could just call <i>scheduler.step()</i> instead of <i>optimizer.step()</i> and go home.

**Comment by SsnL** [comment]
LR schedulers' `step` should be called every epoch, while optimizers' should be called every iteration. The doc is pretty clear on this.

**Comment by kuraga** [comment]
> LR schedulers' `step` should be called every epoch, while optimizers' should be called every iteration. The doc is pretty clear on this.

On 2.2.0, docs don't mention `optimizer.step()` at all (see the [Example](https://pytorch.org/docs/2.2/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html#torch.optim.lr_scheduler.ReduceLROnPlateau)).

Moreover I don't see the `.step(...)` method documentation and/or signature either this LR scheduler nor others.
